dev/src/serializer.ts

```diff
@@ -369,6 +381,8 @@ export function validateUserInput(
     // Ok.
   } else if (value instanceof Timestamp || value instanceof Date) {
     // Ok.
+  } else if (isMomentJsType(value)) {
```

Comment:
Just a thought -- is there value in switching the logic here to use switch/case instead of if/else blocks?

Reply:
As discussed, this doesn't make sense here since we mix "instanceof" with "typeof" among other things.